### PR TITLE
Active enrolment enable work

### PR DESCRIPTION
**Pull request by akssinhaamit**
# Motivation and Context

As per the new design described [here](https://github.com/ONSdigital/ras-rm-documentation/tree/master/service-design/improvements/action-service-restructure-design), this PR makes case to use active enrolment instead of actionPlanId which is behind the feature flag.

# How to test?
Smoke Test

# Links

[Trello](https://trello.com/c/jpQWWct4/246-s24-use-active-enrolments-instead-of-action-plan)

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/casesvc/definition/CaseCreation.java [hunk 1]
```diff
@@ -57,9 +59,9 @@
 
   @XmlElement(required = true)
   protected String sampleUnitType;
-
-  @XmlElement(required = true)
+  @XmlElement(nillable = true)
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/definition/CaseCreation.java:62]: Can you tidy up the spacing?  It's a bit hard to read that actionplanid is nillable.  There should also be a space below that for the activeenrolment variable too

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/Case.java [hunk 2]
```diff
@@ -88,9 +88,12 @@
   @Column(name = "collectioninstrumentid")
   private UUID collectionInstrumentId;
 
-  @Column(name = "actionplanid")
+  @Column(name = "actionplanid", nullable= true)
   private UUID actionPlanId;
 
+  @Column(name = "active_enrolment")
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/Case.java:94]: Does this element also need to be nullable?  What happens if this isn't included (like it is in the current version with the actionplanid)

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/notification/CaseNotification.java [hunk 3]
```diff
@@ -41,10 +41,11 @@
 
   @XmlElement(required = true)
   protected String caseId;
-
-  @XmlElement(required = true)
+  @XmlElement(nillable = true)
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/message/notification/CaseNotification.java:44]: Add a space above this to make it easier to read

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -92,6 +92,7 @@ internet-access-code-svc:
     password: secret
 
 action-svc:
+  deprecated: true
```
Comment (insacuri) [inline src/main/resources/application.yml:95]: the default for now should be false

File: src/main/resources/logback.xml [hunk 5]
```diff
@@ -65,7 +65,7 @@
       </providers>
     </encoder>
     <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
-      <level>DEBUG</level>
+      <level>INFO</level>
```
Comment (insacuri) [inline src/main/resources/logback.xml:68]: Why was this changed?  This'll make it so that case only logs info lines right?  If so, we want it to log at debug level so we can have more and filter out what we don't need

File: src/test/java/uk/gov/ons/ctp/response/casesvc/CaseCreator.java [hunk 6]
```diff
@@ -44,7 +44,7 @@ public CaseNotification sendSampleUnit(
     SampleUnitParent sampleUnit = new SampleUnitParent();
     sampleUnit.setCollectionExerciseId(collectionExerciseId.toString());
     sampleUnit.setId(sampleUnitId.toString());
-    sampleUnit.setActionPlanId(UUID.randomUUID().toString());
+    //sampleUnit.setActionPlanId(UUID.randomUUID().toString());
```
Comment (insacuri) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/CaseCreator.java:47]: Remove if not needed

File: src/test/resources/application-test.yml [hunk 7]
```diff
@@ -12,6 +12,7 @@ survey-svc:
     retry-attempts: 5
 
 action-svc:
+  deprecated: true
```
Comment (insacuri) [inline src/test/resources/application-test.yml:15]: I think this should be false?  Not sure

File: src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java [hunk 8]
```diff
@@ -432,23 +439,34 @@ private void processActionPlanChange(final Case targetCase, final boolean enrolm
...
+          }
+          for (Case caze : cases) {
+            if (caze.getSampleUnitType() == SampleUnitType.B) {
+              caze.setActionPlanId(actionPlans.get(0).getId());
+              caseRepo.saveAndFlush(caze);
+              notificationPublisher.sendNotification(
+                      prepareCaseNotification(caze, CaseDTO.CaseEvent.ACTIONPLAN_CHANGED));
+            }
+          }
+        }else {
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java:462]: spacing here is wrong

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/Case.java (hunk 2 above)
Comment (akssinhaamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/Case.java:94]: Should default to false as its boolen

File: src/main/resources/application.yml (hunk 4 above)
Comment (akssinhaamit) [reply src/main/resources/application.yml:95]: 👍 .

File: src/main/resources/logback.xml (hunk 5 above)
Comment (akssinhaamit) [reply src/main/resources/logback.xml:68]: It is on DEBUG level which makes the log size really big on GCP for devs. We should only make it debug when it is required. It was wrong so changed it to info

File: src/test/java/uk/gov/ons/ctp/response/casesvc/CaseCreator.java (hunk 6 above)
Comment (akssinhaamit) [reply src/test/java/uk/gov/ons/ctp/response/casesvc/CaseCreator.java:47]: Sure.. But If I remove it I can't mark the test config deprecated as true. I will uncomment it for now.

File: src/test/resources/application-test.yml (hunk 7 above)
Comment (akssinhaamit) [reply src/test/resources/application-test.yml:15]: 👍

File: src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java (hunk 8 above)
Comment (akssinhaamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java:462]: 👍

File: src/main/java/uk/gov/ons/ctp/response/casesvc/definition/CaseCreation.java (hunk 1 above)
Comment (akssinhaamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/definition/CaseCreation.java:62]: 👍

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/notification/CaseNotification.java (hunk 3 above)
Comment (akssinhaamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/message/notification/CaseNotification.java:44]: 👍

File: src/test/java/uk/gov/ons/ctp/response/casesvc/CaseCreator.java (hunk 6 above)
Comment (akssinhaamit) [reply src/test/java/uk/gov/ons/ctp/response/casesvc/CaseCreator.java:47]: 👍

**Review by jterryons — APPROVED** [review]
